src/app/pages/wallets/wallet-details/pages-wallet-details-item.model.ts

```diff
@@ -1,17 +1,37 @@
-import { IWalletTransactionApiResponse, WalletTransactionType } from 'src/app/domains/wallets/domains.wallets.types';
+import { TServerEntityId } from 'src/app/common/http/common.http.types';
+import {
+  WalletTransactionType,
+} from 'src/app/domains/transactions/domains.transactions.constants';
+import {
+  IWalletTransactionApiResponse,
+  IWalletTransactionItemData,
+} from 'src/app/domains/transactions/domains.transactions.types';
 
 export class WalletsDetailsTransaction {
```

Comment:
This class is missing `walletId`. Otherwise you won't be able to create new transaction or update existing one - server will return error https://bycza-zagroda.github.io/track-expenses-app-documentation/#/Transactions/createTransaction

You can also create `toPayload(walletId)` method instead and add wallet id to payload object. Up to you.